src/Runner.Listener/Runner.cs

```diff
@@ -733,10 +788,17 @@ await configUpdater.UpdateRunnerConfigAsync(
 
                     if (settings.Ephemeral && runOnceJobCompleted)
                     {
-                        var configManager = HostContext.GetService<IConfigurationManager>();
                         configManager.DeleteLocalRunnerConfig();
                     }
                 }
+
+                // After cleanup, check if we need to restart the session
+                if (restartSession)
+                {
+                    Trace.Info("Restarting runner session after config update...");
+                    RunnerSettings newSettings = configManager.LoadSettings();
+                    return await RunAsync(newSettings, runOnce);
```

Comment:
i am not sure we should do this recursive call, we never did this in the runner before, might be risky.